app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb

```diff
@@ -6,8 +6,8 @@ module IdentityProviders
       # Fetch the identity path from the JWT authenticator policy of the host identity
       FetchIdentityPath = CommandClass.new(
```

Comment:
i would change this class to the following code (let me know if you have any questions):
```
require 'command_class'

module Authentication
  module AuthnJwt
    module IdentityProviders
      # Fetch the identity path from the JWT authenticator policy of the host identity
      FetchIdentityPath = CommandClass.new(
        dependencies: {
          check_authenticator_secret_exists: Authentication::Util::CheckAuthenticatorSecretExists.new,
          fetch_authenticator_secrets:       Authentication::Util::FetchAuthenticatorSecrets.new,
          logger:                            Rails.logger
        },
        inputs:       %i[authentication_parameters]
      ) do
        def call
          fetch_identity_path
        end

        private

        def fetch_identity_path
          @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingIdentityPath.new)
          identity_path
        end

        def identity_path
          return @identity_path if @identity_path

          if identity_path_resource_exists?
            @logger.info(
              LogMessages::Authentication::AuthnJwt::RetrievedResourceValue.new(
                identity_path_secret_value,
                IDENTITY_PATH_RESOURCE_NAME
              )
            )
            @identity_path = identity_path_from_policy
          else
            @logger.debug(
              LogMessages::Authentication::AuthnJwt::IdentityPathNotConfigured.new(
                IDENTITY_PATH_RESOURCE_NAME
              )
            )
            @identity_path = IDENTITY_PATH_DEFAULT_VALUE
          end

          @logger.info(LogMessages::Authentication::AuthnJwt::FetchedIdentityPath.new(@identity_path))
          @identity_path
        end

        def identity_path_resource_exists?
          return @identity_path_resource_exists if defined?(@identity_path_resource_exists)

          @identity_path_resource_exists ||= @check_authenticator_secret_exists.call(
            conjur_account:     @authentication_parameters.account,
            authenticator_name: @authentication_parameters.authenticator_name,
            service_id:         @authentication_parameters.service_id,
            var_name:           IDENTITY_PATH_RESOURCE_NAME
          )
        end

        def identity_path_from_policy
          @identity_path_from_policy ||= identity_path_secret_value
        end

        def identity_path_secret_value
          @identity_path_secret_value ||= @fetch_authenticator_secrets.call(
            conjur_account:          @authentication_parameters.account,
            authenticator_name:      @authentication_parameters.authenticator_name,
            service_id:              @authentication_parameters.service_id,
            required_variable_names: [IDENTITY_PATH_RESOURCE_NAME]
          )[IDENTITY_PATH_RESOURCE_NAME]
        end
      end
    end
  end
end
```